Sources/MapboxCoreNavigation/NavigationSettings.swift

```diff
@@ -23,6 +24,47 @@ public class NavigationSettings {
             }
         }
     }
+
+    private let lock: NSLock = .init()
+    private var _directions: Directions?
+    private var _tileStoreConfiguration: TileStoreConfiguration?
```

Comment:
I would suggest creating a struct with these two properties and store that struct. This reduces code duplication in accessors.

Reply:
Done, please review the latest revision.